### DISCUSSION TITLE
clamav arm64

### DISCUSSION THREAD

**Discussion by Zoey2936** [nextcloud/all-in-one#4223] [Ideas] [answered]
clamav-docker supports arm64, see: https://hub.docker.com/r/clamav/clamav-debian/tags
or clamav for arm64 could be pulled from alpine like mailcow does: https://github.com/mailcow/mailcow-dockerized/blob/master/data/Dockerfiles/clamd/Dockerfile

**Comment by szaimen** [comment]
Hi, thanks for the idea! 

However the best solution to me seems that clamav at some point officially ships multiarch alpine based images which they still do not... So I guess opening another feature request in their repo would make sense?

**Reply by Zoey2936** [reply]
https://github.com/Cisco-Talos/clamav-docker/issues/8 they closed and said arm64 exists for debian, so noieda

**Reply by szaimen** [reply]
wrote a comment

**Reply by jogerj** [reply]
the [clamav](https://hub.docker.com/r/clamav/clamav/tags) image is already based on alpine

**Reply by Zoey2936** [reply]
But does not support arm64

**Reply by jogerj** [reply]
ah I missed that bit. The debian builds are indeed cross platform but alpine builds are only amd64, as specified from the Jenkinsfile in the repo. I digged through old issues and there's a comment https://github.com/Cisco-Talos/clamav-docker/issues/5#issuecomment-1371531980 where the main image could be debian instead of alpine in the future... I've opened another issue https://github.com/Cisco-Talos/clamav-docker/issues/48

Edit: it seems the answer from maintainers were there are no plans to add alpine arm64 images. We'll have to use the debian images or build our own. Debian images risks transitive vulnerabilities, mostly from glibc (where alpine uses musl) and other bundled packages.

That said taking the docker setup from mailcow is a valid option (mailcow is GPL and compatible with AGPL) but risks maintenance effort in NC

**Reply by Zoey2936** [reply]
I think the best way would be to do it like mailcow and pull clamav as an alpine package

**Reply by szaimen** [reply]
Yeah, you are probably right. Can you  create a PR? :)

**Reply by Zoey2936** [reply]
when I have time, yes

**Reply by szaimen** [reply, marked as answer]
Will be done in https://github.com/nextcloud/all-in-one/pull/6094

**Reply by szaimen** [reply]
This is now released with v10.8.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
